cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -650,6 +662,12 @@ public void deleteApplications(String namespaceId)
     getApplicationSpecificationTable().deleteAll(getNamespaceRange(namespaceId));
   }
 
+  public void markDeleteApplication(ApplicationReference appRef) throws IOException {
+    Collection<Field<?>> fields =
+      ImmutableList.of(Fields.stringField(StoreDefinition.AppMetadataStore.STATUS_FIELD, Status.DELETED.toString()));
+    getApplicationSpecificationTable().updateAll(getNamespaceAndApplicationRange(appRef), fields);
```

Comment:
this is updating all the rows in the table? I don't think this will be any faster than just deleting them. 

The benefit of marking is if it only involves a few rows and not an unbounded number. For example, does it work if we just mark the latest one as deleted?

Reply:
I think we can check for active status only on the latest version, in that case we need to mark only the latest version